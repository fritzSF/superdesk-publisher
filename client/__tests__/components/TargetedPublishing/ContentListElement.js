import React from "react";
import ContentListElement from "../../../components/TargetedPublishing/ContentListElement";
import { render, fireEvent } from "@testing-library/react";

describe("TargetedPublishing/ContentListElement", () => {
  const ruleLists = [
    {
      id: 1,
      position: 0
    }
  ];

  const list = {
    id: 1,
    position: 0
  };

  const allContentLists = [
    {
      content_list_items_count: 10,
      id: 1,
      name: "test1"
    },
    {
      content_list_items_count: 5,
      id: 2,
      name: "test2"
    }
  ];

  it("renders correctly", () => {
    const { container } = render(
      <ContentListElement
        removeList={jest.fn()}
        save={jest.fn()}
        index={0}
        ruleLists={ruleLists}
        list={list}
        allContentLists={allContentLists}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correct number of content lists to select", () => {
    const { container } = render(
      <ContentListElement
        removeList={jest.fn()}
        save={jest.fn()}
        index={0}
        ruleLists={ruleLists}
        list={list}
        allContentLists={allContentLists}
      />
    );

    const select = container.querySelector('select[name="id"]');

    // 2 content lists + 1 empty option = 3
    expect(select.childElementCount).toBe(3);
  });

  it("renders correct number of position options", () => {
    const { container } = render(
      <ContentListElement
        removeList={jest.fn()}
        save={jest.fn()}
        index={0}
        ruleLists={ruleLists}
        list={list}
        allContentLists={allContentLists}
      />
    );

    const select = container.querySelector('select[name="position"]');

    expect(select.childElementCount).toBe(10);
  });

  it("fires removeList with proper arguments", () => {
    const removeList = jest.fn();

    const { container } = render(
      <ContentListElement
        removeList={removeList}
        save={jest.fn()}
        index={0}
        ruleLists={ruleLists}
        list={list}
        allContentLists={allContentLists}
      />
    );

    const button = container.querySelector('a.icn-btn .icon-trash').closest('a.icn-btn');

    fireEvent.click(button);

    expect(removeList).toHaveBeenCalled();
    expect(removeList).toBeCalledWith(0);
  });

  it("fires save with proper arguments", () => {
    const save = jest.fn();

    const { container, getByText } = render(
      <ContentListElement
        removeList={jest.fn()}
        save={save}
        index={0}
        ruleLists={ruleLists}
        list={list}
        allContentLists={allContentLists}
      />
    );

    const select = container.querySelector('select[name="position"]');

    fireEvent.change(select, { target: { value: 4 } });

    expect(save).toHaveBeenCalled();
    expect(save).toBeCalledWith([
      {
        id: 1,
        position: 4
      }
    ]);
  });
});
