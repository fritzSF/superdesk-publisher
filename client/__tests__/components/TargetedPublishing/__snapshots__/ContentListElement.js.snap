// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TargetedPublishing/ContentListElement renders correctly 1`] = `
<div
  class="sd-list-item sd-list-item--no-hover sd-margin-b--1"
>
  <div
    class="sd-list-item__column sd-list-item__column--grow sd-list-item__column--large-padding"
  >
    <div
      class="sd-list-item__row no-margin sd-flex-justify-end"
    >
      <button
        class="icn-btn"
        tabindex="0"
      >
        <i
          aria-hidden="true"
          aria-label="trash"
          class="icon-trash"
        />
      </button>
    </div>
    <div
      class="sd-list-item__row"
    >
      <div
        class="sd-line-input sd-line-input--is-select sd-list-item--element-grow"
      >
        <label
          class="sd-line-input__label"
        >
          Choose list
        </label>
        <select
          class="sd-line-input__select"
          name="id"
        >
          <option
            value=""
          />
          <option
            value="1"
          >
            test1
          </option>
          <option
            value="2"
          >
            test2
          </option>
        </select>
      </div>
    </div>
    <div
      class="sd-list-item__row"
    >
      <div
        class="sd-line-input sd-line-input--is-select sd-list-item--element-grow"
      >
        <label
          class="sd-line-input__label"
        >
          Set order
        </label>
        <select
          class="sd-line-input__select"
          name="position"
        >
          <option
            value="0"
          >
            1
          </option>
          <option
            value="1"
          >
            2
          </option>
          <option
            value="2"
          >
            3
          </option>
          <option
            value="3"
          >
            4
          </option>
          <option
            value="4"
          >
            5
          </option>
          <option
            value="5"
          >
            6
          </option>
          <option
            value="6"
          >
            7
          </option>
          <option
            value="7"
          >
            8
          </option>
          <option
            value="8"
          >
            9
          </option>
          <option
            value="9"
          >
            10
          </option>
        </select>
      </div>
    </div>
  </div>
</div>
`;
